integrations/nvidia/src/haystack_integrations/components/rankers/nvidia/ranker.py

```diff
@@ -0,0 +1,247 @@
+#   Deserializes the ranker from a dictionary.
+#
+# - `run(self,
+#        query: str,
+#        documents: List[Document],
+#        top_k: Optional[int]) -> List[Document]`
+#   Ranks a list of documents based on a given query.
+#
+# The class should have the following attributes:
+#
+# - `model: str`
+#   The model to use for ranking.
+#
+# - `api_url: str`
+#   The URL of the API.
+#
+# - `api_key: str`
+#   The API key.
+#
+# - `truncate: Optional[Literal["NONE", "END"]]`
+#   The truncation strategy to use.
+#
+
+_DEFAULT_MODEL = "nvidia/nv-rerankqa-mistral-4b-v3"
+
+_MODEL_ENDPOINT_MAP = {
+    "nvidia/nv-rerankqa-mistral-4b-v3": "https://ai.api.nvidia.com/v1/retrieval/nvidia/nv-rerankqa-mistral-4b-v3/reranking",
+}
+
+
+@component
+class NvidiaRanker:
+    """
+    A component for ranking documents using ranking models provided by
+    [NVIDIA NIMs](https://ai.nvidia.com).
+
+    Usage example:
+    ```python
+    from haystack_integrations.components.rankers.nvidia import NvidiaRanker
+
+    ranker = NvidiaRanker(
+        model="nvidia/nv-rerankqa-mistral-4b-v3",
+        api_key="your-api-key",
+    )
+    ranker.warm_up()
+
+    query = "What is the capital of Germany?"
+    documents = [
+        Document(content="Berlin is the capital of Germany."),
+        Document(content="The capital of Germany is Berlin."),
+        Document(content="Germany's capital is Berlin."),
+    ]
+
+    result = ranker.run(query, documents, top_k=2)
+    print(result["documents"])
+    ```
+    """
+
+    def __init__(
+        self,
+        model: Optional[str] = _DEFAULT_MODEL,
+        truncate: Optional[Literal["NONE", "END"]] = None,
+        api_url: Optional[str] = None,
+        api_key: Optional[Secret] = None,
+        top_k: int = 5,
+    ):
+        """
+        Create a NvidiaRanker component.
+
+        :param model:
+            Ranking model to use.
+        :param truncate:
+            Truncation strategy to use. Can be "NONE" or "END". Defaults to NIM's default.
+        :param api_key:
+            API key for the NVIDIA NIM.
+        :param api_url:
+            Custom API URL for the NVIDIA NIM.
+        :param top_k:
+            Number of documents to return.
+        """
+        if not isinstance(model, str):
+            msg = "Ranker expects the `model` parameter to be a string."
+            raise TypeError(msg)
+        if not isinstance(api_url, (str, type(None))):
+            msg = "Ranker expects the `api_url` parameter to be a string."
+            raise TypeError(msg)
+        if truncate is not None and truncate not in ["NONE", "END"]:
+            msg = "Invalid truncate value. Must be 'NONE' or 'END'."
+            raise ValueError(msg)
+        if not isinstance(top_k, int) or isinstance(top_k, bool):
+            msg = "Ranker expects the `top_k` parameter to be an integer."
+            raise TypeError(msg)
+
+        self._model = model
+        self._truncate = truncate
+        self._api_key = api_key
```

Comment:
we dropped the use of plain text API, keys, you can simply to this, and the user sets it with an environment variable  or pass with this parameter

```suggestion
        self._api_key = Secret.from_env_var("NVIDIA_API_KEY")
```

Reply:
should i entirely remove the `api_key` param to `__init__`?